src/analysis/liveness-transfer-function.h

```diff
@@ -0,0 +1,101 @@
+#ifndef wasm_analysis_liveness_transfer_function_h
+#define wasm_analysis_liveness_transfer_function_h
+
+#include "lattice.h"
+#include "monotone-analyzer.h"
+
+namespace wasm::analysis {
+
+class LivenessTransferFunction : public Visitor<LivenessTransferFunction> {
+  FinitePowersetLattice::Element currState;
```

Comment:
`currState` is always reset each time the transfer function is applied. It is a member variable to allow the visitor functions to access it.

Reply:
Since it's logically owned by the core analysis code rather than the transfer function, perhaps it would make sense to make `currState` a pointer to the element in the analysis code rather than a separate value that has to be copied into and out of. WDYT?

Reply:
Good idea, done.